alicloud/resource_alicloud_mns_queue.go

```diff
@@ -0,0 +1,187 @@
+	if err != nil {
+		return fmt.Errorf(" creating alicoudMNSQueue  error: %#v", err)
+	}
+	queueManager := ali_mns.NewMNSQueueManager(*mnsClient)
+	attr, err := queueManager.GetQueueAttributes(d.Id())
+	if err != nil {
+		return err
+	}
+	d.Set("name", attr.QueueName)
+	d.Set("delay_seconds", attr.DelaySeconds)
+	d.Set("maximum_message_size", attr.MaxMessageSize)
+	d.Set("message_retention_period", attr.MessageRetentionPeriod)
+	d.Set("visibility_timeout", attr.VisibilityTimeout)
+	d.Set("polling_wait_seconds", attr.PollingWaitSeconds)
+
+	return nil
+}
+
+func resourceAlicloudMNSQueueUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	mnsClient, err := client.Mnsconn()
+	if err != nil {
+		return fmt.Errorf(" creating alicoudMNSQueue  error: %#v", err)
+	}
+	queueManager := ali_mns.NewMNSQueueManager(*mnsClient)
+	d.Partial(true)
+	attributeUpdate := false
+
+	attr, err1 := queueManager.GetQueueAttributes(d.Get("name").(string))
+	if err1 != nil {
+		return err1
+	}
+	var delaySeconds, maximumMessageSize, messageRetentionPeriod, visibilityTimeouts, pollingWaitSeconds int
+	delaySeconds = int(attr.DelaySeconds)
+	maximumMessageSize = int(attr.MaxMessageSize)
+	messageRetentionPeriod = int(attr.MessageRetentionPeriod)
+	visibilityTimeouts = int(attr.VisibilityTimeout)
+	pollingWaitSeconds = int(attr.PollingWaitSeconds)
+	name := d.Get("name").(string)
+	if d.HasChange("delay_seconds") {
+		d.SetPartial("delay_seconds")
+		delaySeconds = d.Get("delay_seconds").(int)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("maximum_message_size") {
+		d.SetPartial("maximum_message_size")
+		maximumMessageSize = d.Get("maximum_message_size").(int)
+		attributeUpdate = true
+	}
+
+	if d.HasChange("message_retention_period") {
+		d.SetPartial("message_retention_period")
+		messageRetentionPeriod = d.Get("message_retention_period").(int)
+		attributeUpdate = true
+	}
+	if d.HasChange("visibility_timeout") {
+		d.SetPartial("visibility_timeout")
+		visibilityTimeouts = d.Get("visibility_timeout").(int)
+		attributeUpdate = true
+	}
+	if d.HasChange("polling_wait_seconds") {
+		d.SetPartial("polling_wait_seconds")
+		pollingWaitSeconds = d.Get("polling_wait_seconds").(int)
+		attributeUpdate = true
+	}
+	d.Partial(false)
+
+	if attributeUpdate {
+		err = queueManager.SetQueueAttributes(name, int32(delaySeconds), int32(maximumMessageSize), int32(messageRetentionPeriod), int32(visibilityTimeouts), int32(pollingWaitSeconds), 3)
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func resourceAlicloudMNSQueueDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	mnsClient, err := client.Mnsconn()
+	if err != nil {
+		return fmt.Errorf(" creating alicoudMNSQueue  error: %#v", err)
+	}
+	queueManager := ali_mns.NewMNSQueueManager(*mnsClient)
+	name := d.Get("name").(string)
+	err = queueManager.DeleteQueue(name)
+	if err != nil {
+		return err
+	}
```

Comment:
Please describe queue to ensure it has been deleted truly after running `deletequeue`.